source/codegen/service_helpers.py

```diff
@@ -130,47 +146,50 @@ def get_array_param_size(parameter) -> str:
 def expand_varargs_parameters(parameters):
     if not common_helpers.has_repeated_varargs_parameter(parameters):
         return parameters
-    # omit the varargs parameter that we're going to expand
-    new_parameters = parameters[:-1]
-    varargs_parameter = parameters[-1]
-    assert common_helpers.is_repeated_varargs_parameter(varargs_parameter)
-    max_length = varargs_parameter['max_length']
-    # Many (all?) functions that take varargs take the first set of parameter as
-    # non-varargs. If this is the case, we need one fewer set of parameters in the varargs
+    # omit the varargs parameters that we're going to expand
+    new_parameters = [
+        p for p in parameters if not common_helpers.is_repeated_varargs_parameter(p)]
+    varargs_parameters = [
+        p for p in parameters if common_helpers.is_repeated_varargs_parameter(p)]
+    max_length = varargs_parameters[0]['max_length']
+    # All functions that take repeated varargs take the first set of parameter as
+    # non-varargs, so we need one fewer set of parameters in the varargs
     # section.
-    if any([p for p in parameters if not p.get('include_in_proto', True)]):
-        max_length -= 1
+    max_length -= 1
+    repeated_parameters = [
+        p for p in parameters if common_helpers.is_repeating_parameter(p)]
     for i in range(max_length):
-        for field in varargs_parameter['varargs_type']['fields']:
-            new_parameters.append({'cppName': f'{field["name"]}{i}'})
+        for p in repeated_parameters:
+            new_parameters.append({'cppName': f'{p["name"]}{i}'})
     return new_parameters
 
 
-def create_param(parameter, expand_varargs=True, any_not_include_in_proto=False):
+def create_param(parameter, expand_varargs=True, repeated_parameters=None):
     type = parameter['type']
     name = parameter['cppName']
     if common_helpers.is_struct(parameter):
         type = type.replace("struct ", "")
-    if common_helpers.is_array(type):
-        array_size = get_array_param_size(parameter)
-        return f'{type[:-2]} {name}[{array_size}]'
-    elif common_helpers.is_pointer_parameter(parameter):
-        return f'{type}* {name}'
-    elif common_helpers.is_repeated_varargs_parameter(parameter):
+    if common_helpers.is_repeated_varargs_parameter(parameter):
         if expand_varargs:
-            max_length = parameter['max_length']
-            if any_not_include_in_proto:
-                max_length -= 1
+            # All functions that take repeated varargs take the first set of parameter as
+            # non-varargs, so we need one fewer set of parameters in the varargs
+            # section.
+            max_length = parameter['max_length'] - 1
```

Comment:
Consider: this shows up twice with the same block comment. Can it be a helper function somewhere? get_expanded_var_args_length(param)?

Reply:
This went away since now we're passing an extra set of arguments.